scheduler/src/cook/mesos/util.clj

```diff
@@ -209,16 +242,21 @@
          ;; the total number of jobs for a user should be smaller than
          ;; all completed jobs by any user i.e.
          ;; (waiting_user + running_user < completed - completed_user)
-         (q '[:find [?j ...]
-              :in $ ?user ?state ?start ?end
-              :where
-              [?j :job/user ?user]
-              [?j :job/state ?state]
-              [?j :job/submit-time ?t]
-              [(< ?start ?t)]
-              [(< ?t ?end)]
-              [?j :job/custom-executor false]]
-            db user state start end)
+
+         (mapcat (fn query-in-chunks
+                   [interval]
+                   (let [[interval-start interval-end] (map tc/to-date interval)]
+                     (q '[:find [?j ...]
+                          :in $ ?user ?state ?start ?end
+                          :where
+                          [?j :job/submit-time ?t]
+                          [(< ?start ?t)]
+                          [(< ?t ?end)]
+                          [?j :job/user ?user]
+                          [?j :job/state ?state]
+                          [?j :job/custom-executor false]]
+                        db user state interval-start interval-end)))
+                 (generate-intervals (tc/from-date start) (tc/from-date end)))
```

Comment:
Should returning the appropriate date objects be the responsibility of `generate-intervals` so we can avoid the calls to `tc/from-date` and `tc/to-date`?